### PR TITLE
[12.0][ADD] account_move_ref_document

### PR DESCRIPTION
**Pull request by kittiu**
WIP, just for discussion.

The objective of this module is to create permanent link from journal entries back to
the source document that create it. For example, validating invoice creates journal entry. Invoice will be
source document, and journal entry will have permanent link back its source.

This module add new field Document Ref to Journal Entry, as ``document_id``, ``document_ref``
to account.move and ``document_id`` to account.move.line.

After installing this module, following modules that helps calculate the link will be auto installed,

- account_move_ref_account
- account_move_ref_expense

Currently the supported documents are account.invoice, account.payment, account.bank.statement and hr.expense.sheet

### CODE REVIEW

**Comment by kittiu** [comment]
Before further proceeding. Any comment or disagreement on doing this module are welcome.

File: account_move_ref_account/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    'name': 'Account Move Source Document for Invoice/Payment/Statement',
+    'summary': 'Journal Entry with Source Document Link',
+    'version': '12.0.1.0.0',
+    'author': 'Ecosoft,Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/account-financial-tools',
+    'category': 'Accounting & Finance',
+    'depends': ['account_move_ref_document',
+                'account'],
+    'data': [],
```
Comment (nikul-serpentcs) [inline account_move_ref_account/__manifest__.py]: @kittiu If not any `file` then remove `data`

File: account_move_ref_document/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from odoo import api, models, fields
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    document_id = fields.Reference(
+        selection='_selection_model',
+        string='Document',
+        help="Reference to the document that create this account.move",
```
Comment (nikul-serpentcs) [inline account_move_ref_document/models/account_move.py]: `Reference to the document that create this Account Move`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
netpicking

File: account_move_ref_document/models/account_move.py (hunk 2 above)
Comment (kittiu) [reply account_move_ref_document/models/account_move.py]: Done

File: account_move_ref_account/__manifest__.py (hunk 1 above)
Comment (kittiu) [reply account_move_ref_account/__manifest__.py]: Done

**Comment by kittiu** [comment]
This module may be overkilled. So, I decide to close it.

**Pull request closed without merging**